conda/repodata.py

```diff
@@ -0,0 +1,99 @@
+# (c) Continuum Analytics, Inc. / http://continuum.io
+# All Rights Reserved
+#
+# conda is distributed under the terms of the BSD 3-clause license.
+# Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from .core import repodata as _impl
+from .connection import CondaSession
+from os.path import join
+
+
+class RepoData(object):
+    """This object represents all the package metainfo of a single channel."""
+
+    _data = {}
+
+    @staticmethod
+    def enable(url, name, priority, cache_dir=None):
+        RepoData._data[url] = RepoData(url, name, priority, cache_dir)
+
```

Comment:
I'm not sure I understand. At some place some code needs to explicitly instantiate RepoData objects from channel URLs (likely the ones a user has "enabled" in his ".condarc" file. The `enable` method does just that.
The `get` then is nothing but a simple way to access individual RepoData objects once created. It isn't meant to magically instantiate anything.
Note that I imagine that we do want the ability to access RepoData objects separately (in contrast to current conda code that only uses a single "index" dict that has everything merged into it).